### ISSUE TITLE
[2.0.x] Setting form element name to "value" produces BadMethodCallException during render

### ISSUE THREAD

**Issue by cinu** [phalcon/cphalcon#10398]
Setting form element name to "value" produces BadMethodCallException  (Wrong number of parameters) during render.

Reproduction code:

```php
<?php
$di = new \Phalcon\DI\FactoryDefault();
 
use Phalcon\Forms\Form;
use Phalcon\Forms\Element\Text;
 
class FormTest extends Form
{
    public function initialize()
    {
        $this->add(new Text("value"));
        $this->add(new Text("value2"));
    }
}
 
$f=new FormTest();
echo $f->render('value2'); // works
echo $f->render('value'); // exception
 
/*
output:
 
<input type="text" id="value2" name="value2" />PHP Fatal error:  Uncaught exception 'BadMethodCallException' with message 'Wrong number of parameters' in /.../test.php:18
Stack trace:
#0 [internal function]: Phalcon\Forms\Form->getValue()
#1 [internal function]: Phalcon\Forms\Form->getValue('value')
#2 [internal function]: Phalcon\Forms\Element->getValue()
#3 [internal function]: Phalcon\Forms\Element->prepareAttributes(NULL)
#4 [internal function]: Phalcon\Forms\Element\Text->render(NULL)
#5 /.../test.php(18): Phalcon\Forms\Form->render('value')
#6 {main}
  thrown in /.../test.php on line 18
```

PHP: 5.6.7-1 (debian)
Phalcon: 2.0.1

**Comment by jkmf** [comment]
I just like to add that other forbidden form element names are:
'label', 'messages', 'messagesfor', 'elements' 
and probably lot more...

**Comment by berliner** [comment]
Elements with name "value" are getting rendered without problems wenn passing a model into the form constructor, e.g.:
`$form = new Form(new MyModel());`

Tested with phalcon 2.0.1

**Comment by andresgutierrez** [comment]
It checks if the form has a 'getter' to obtain the value of the element, as the form value has a method called getValue, it tries to use it to obtain the value

**Comment by mzf** [comment]
+1 
If column named as `value` then `Form\Element` throw `Wrong number of parameters` when call `render`

**Comment by sergeyklay** [comment]
I'll try to sort out

**Comment by sergeyklay** [comment]
Fixed in the `2.1.x` branch.
